### PR TITLE
Fix parent methods in qobject

### LINKED ISSUE #1829: Cleanup parent()/setParent() methods for classes derived from QObject
Following-up investigations about #1824, a number of classes, including WbNode and WbField inherit from QObject and implement parent() and setParent() methods with a different signature from the one of QObject. This is highly error prone and should be fixed by renaming these methods. Moreover, WbField.cpp calls the setParent() method of QObject which is likely wrong...

### PR DESCRIPTION
**Pull request by omichel**
Fixed #1829, including the removal of apparently useless calls to QObject::setParent() in WbField.cpp.
✔️ passed the test suite locally on Windows.

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:eyes:

File: src/webots/vrml/WbNode.cpp [hunk 1]
```diff
@@ -827,7 +827,7 @@ WbField *WbNode::findSubField(int index, WbNode *&parentNode) const {
   foreach (WbNode *const node, list) {
     foreach (WbField *const field, node->mFields) {
       if (count == index) {
-        parentNode = node;
+        parent = node;
```
Comment (DavidMansolino) [inline src/webots/vrml/WbNode.cpp:830]:
```suggestion
        parentNode = node;
```
This change is wrong.

Comment (omichel) [reply src/webots/vrml/WbNode.cpp:830]: Well spotted! But in that case, I would prefer to rename the parameter name as the `parentNode` parameter name shadows the `parentNode()` method.

**Comment by omichel** [comment]
For consistency, I also changed `WbNode::setGlobalParent()` to `WbNode::setGlobalParentNode()`, but there was much more occurrences...

**Pull request merged**